### DISCUSSION TITLE
[Resolved ✅ ] Potential security vulnerability

### DISCUSSION THREAD

**Discussion by Fewword** [BerriAI/litellm#2936] [General]
## Status - Resolved ✅
Few security vulnerability has been disclosed to [huntr](https://huntr.com/) for this repo.
Fix those bug ASAP
report link -->
https://huntr.com/bounties/70897f59-a966-4d93-b71e-745e3da91970

report are currently private and only repo maintainer can see the report details.
If you are repo maintainer then there is 2 way to view the report details

by signup
by secret magick link that has been sent to your mail by huntr.com.

**Comment by ghost** [comment]
Hi @Fewword could you please share the issues? We received a huntr email, but when going to the link they shared, we were unable to create additional keys - as our db auth checks this. 

Open to feedback though. Can you show me how you might create additional keys with the default user flow? 


Context: 
I'm the LiteLLM Maintainer

**Comment by ghost** [comment]
converting to discussion for now

**Comment by Fewword** [comment]
hi @krrishdholakia thanks for the quick response! Just to clarify, have you registered for a Huntr account using your GitHub account? This should give you the access needed to view the details of the security vulnerability I've reported. If there's any confusion or additional steps needed, I'm here to help guide you through the process.

Looking forward to your feedback and resolving the vulnerabilities together!

**Reply by ghost** [reply]
> when going to the link they shared, we were unable to create additional keys - as our db auth checks this.

Can you show me how you might create additional keys with the default user flow?


any help understanding how you would do this, would be great @Fewword

**Reply by Fewword** [reply]
Hi there,

I noticed that a member of your team, ishaan-jaff, has successfully logged into Huntr and acknowledged a vulnerability report submitted earlier. Could you possibly check with him on the specific steps he followed to log in and access the report? It might shed some light on how we can resolve the current access issue you're experiencing with the additional keys.
![WeChate174c65993bec805a43192f950065688](https://github.com/BerriAI/litellm/assets/76860434/1c026245-d3f4-40b1-8ce3-9cac89a48c91)

**Reply by ishaan-jaff** [reply]
@Fewword we've resolved it #2949

**Reply by Fewword** [reply]
@ishaan-jaff While we're on this topic, I wanted to mention that I've submitted a few new vulnerability reports for your project on Huntr. When you have a moment, could you please take a look at these additional reports? Your review and feedback would be highly valuable.
https://huntr.com/bounties/70897f59-a966-4d93-b71e-745e3da91970
https://huntr.com/bounties/77c34da8-3826-4f2a-bf82-efbb9005e91b
https://huntr.com/bounties/8b9039ae-842b-4494-8329-94a4b353f910

**Reply by ishaan-jaff** [reply]
Huntr does not give me access to the reports - followed up on their discord

**Reply by ghost** [reply]
@Fewword curious - why not just file them as issues or dm us on discord, so we can fix it quickly?

**Reply by Fewword** [reply]
@krrishdholakia The reason I'm submitting through Huntr is that it not only provides a structured platform for tracking and acknowledging vulnerabilities but also offers bounties for researchers and project maintainers upon confirmation and resolution of the issues. This system not only incentivizes the security research community but also supports the maintainers for their efforts in resolving these vulnerabilities.

**Reply by Fewword** [reply]
<img width="864" alt="image" src="https://github.com/BerriAI/litellm/assets/76860434/f7a4d69f-aab8-4109-88ef-0855beb79cc2">
is this your discord account? @krrishdholakia

**Reply by ishaan-jaff** [reply]
all the bounties were addressed over huntr @Fewword

**Reply by Fewword** [reply]
hi @ishaan-jaff in my poc 
`curl --location 'http://localhost:4000/team/delete'     --header 'Authorization: Bearer sk-EDPWeidMYhR87NOwUwoR6g'  --header 'Content-Type: application/json'  --data-raw '{ "team_ids": ["45e3e396-ee08-4a61-a88e-16b3ce7e0849"(CHANGE HERE)] }'`
Authorization: Bearer sk-EDPWeidMYhR87NOwUwoR6g this is user api-key and this stands for user1 that admin created

**Reply by Fewword** [reply]
<img width="1190" alt="image" src="https://github.com/BerriAI/litellm/assets/76860434/5abc2de0-1dcb-48ab-8ee2-a9e2bf33afff">

**Reply by ishaan-jaff** [reply]
@Fewword  i'm confused what is your concern here ? 

1. You need to have a valid key to call `/team/delete`

**Reply by Fewword** [reply]
@ishaan-jaff Doesn't every user, whether registered or created by an administrator, have a valid key?

**Reply by ishaan-jaff** [reply]
@Fewword can you explain what is the security vulnerability here ? If you're not authenticated you cannot call `/team/delete`

**Reply by Fewword** [reply]
@ishaan-jaff I ran the litellm docker image locally and then used the default administrator account to create a user named user1. After creation, the api-key provided for user1 was sk-EDPWeidMYhR87NOwUwoR6g. Then, I used the administrator account to create a team called team1, with the team ID being 45e3e396-ee08-4a61-a88e-16b3ce7e0849. However, at that time, team1's members included only the default administrator, and user1 was not a member of team1. When I called the API using the key sk-EDPWeidMYhR87NOwUwoR6g for authentication, the system did not check user1's roles and permissions, lacking authorization verification, which resulted in the deletion of team1.

**Reply by ishaan-jaff** [reply]
okay understood - Tracking that as an issue here and we will fix it: https://github.com/BerriAI/litellm/issues/2955 

@Fewword We would appreciate if you filed these as issues on the litellm Github Repo or on discord, it allows us to address them much faster

**Reply by Fewword** [reply]
@ishaan-jaff No problem, glad to help! Thanks for tracking this as an issue. I'd also appreciate if you could re-evaluate the vulnerability reports I've submitted to ensure they are addressed properly. Looking forward to the fixes and happy to contribute further!

**Reply by Fewword** [reply]
hi @ishaan-jaff, could you update my reports on huntr and validate these vulnerabilities?
